Automattic/wp-calypso
# Prevent ID collisions in posts queries that span multiple sites

I found this issue while investigating #17549 and found it necessary to fix it to test that change sufficiently.

I created a pair of test sites `test1user2sites` and `test1user2sites2`, both owned by the user `test1user2sites`.  Because I created each one using the same steps, they have the same post IDs assigned to their first two posts:  4 and 16.

When rendered into a `PostTypeList`, this is an ID collision, and entries with the same ID are only pulled from 1 site:

<img src="https://user-images.githubusercontent.com/227022/29798647-e09f8002-8c23-11e7-8063-fbd6a78ab2a3.png" width="500">

We can fix this by using `global_ID` as the key for `PostQueryManager`.  Due to uncertainty about the broader effects of making this change for all posts queries, I have only changed the key for queries that involve multiple sites.

<img src="https://user-images.githubusercontent.com/227022/29798716-5782763e-8c24-11e7-9fbb-02989acbc4e9.png" width="500">

### To test

Log in as user `test1user2sites`.

Ensure that the changes from #17549 are applied, whether after its merge or through other means.

Start Calypso using `ENABLE_FEATURES=posts/post-type-list npm start`.

Visit `/posts` and verify that before this PR, the posts list displays broken, duplicated posts as in the first screenshot, and after, the posts from both sites are displayed correctly as in the second screenshot.
